app/scripts/views/force_auth.js

```diff
@@ -32,17 +36,102 @@ define(function (require, exports, module) {
     afterSignInBrokerMethod: 'afterForceAuth',
     afterSignInNavigateData: { clearQueryParams: true },
 
-    context: function () {
-      var fatalError = '';
-      var email = this.relier.get('email');
+    _fatalError: null,
+
+    beforeRender: function () {
+      var self = this;
+      var relier = this.relier;
+
+      if (! relier.has('email')) {
+        this._fatalError = AuthErrors.toError('FORCE_AUTH_EMAIL_REQUIRED');
+        return;
+      }
 
-      if (! email) {
-        fatalError = AuthErrors.toError('FORCE_AUTH_EMAIL_REQUIRED');
+      /**
+       * If the relier specifies a UID, check whether the UID is still
+       * registered. If the uid is no longer registered, notify the
+       * broker that the account has been deleted. The user will
+       * still be allowed to sign in with their email address. If
+       * no account is registered with the given email address,
+       * the broker will decide whether to allow the user to sign up.
+       */
+      var account = this.user.initAccount({
+        email: relier.get('email'),
+        uid: relier.get('uid')
+      });
+
+      if (relier.has('uid')) {
+        // relier specified a uid, if the uid is no longer registered
+        // then the user can only signin/signup if the relier (via the broker)
+        // supports uid changes.
+        return p.all([
+          this.user.checkAccountEmailExists(account),
+          this.user.checkAccountUidExists(account)
+        ]).spread(function (emailExists, uidExists) {
+          /*
+           * uidExists - false, emailExists - false - let user sign up w/ email
+           * uidExists - true, emailExists - false - uid exists but
+           *   doesn't match email, how'd this happen? let the user sign up
+           * uidExists - false, emailExists - true - sign in w/ new uid
+           * uidExists - true, emailExists - true - assume for the same
+           *   account, try to sign in
```

Comment:
For some reason, this comment seems harder to parse than it should be. All those hyphens confuse my eyes/brain. Could we maybe try using a different separator for the values and the description? e.g.:

```js
/*
 * uidExists: false, emailExists: false - let user sign up w/ email
 */
```